spras/netmix2.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+
+from spras.containers import prepare_volume, run_container
+from spras.dataset import Dataset
+from spras.prm import PRM
+
+__all__ = ['NetMix2']
+
+class NetMix2(PRM):
+    required_inputs = ['network', 'scores']
+
+    @staticmethod
+    def generate_inputs(data: Dataset, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: Dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in NetMix2.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing")
+
+        if data.contains_node_columns('prize'):
+            node_df = data.request_node_columns(['prize'])
+        else:
+            raise ValueError("Node prizes are required for NetMix2.")
+        node_df.to_csv(filename_map['scores'], index=False, columns=['prize', 'NODEID'], header=False, sep='\t')
+
+        edges_df = data.get_interactome()
+        edges_df.to_csv(filename_map['network'], index=False, sep='\t', columns=['Interactor1', 'Interactor2'], header=False)
+    
+    @staticmethod
+    def run(network=None, scores=None, output_file=None, delta=None, num_edges=None, density=None,
+            time_limit=None,
```

Comment:
Should we preset a time_limit?

Reply:
I wasn't quite sure how to handle that, as `time_limit` feels like another option that should really be global to algorithms.